[rum] core web vitals (#9737)
DataDog/documentation: content/en/real_user_monitoring/browser/monitoring_page_performance.md

    text: "RUM Dashboards"
---

{{< img src="real_user_monitoring/browser/waterfall.png" alt="Real User Monitoring page load waterfall" style="width:75%;" >}}

## Performance metrics for views

> RUM view events collect extensive performance metrics for every single page view. We recommend analyzing the performance metrics in the following ways:

Reviewer: ```suggestion

RUM view events collect extensive performance metrics for every single page view. Datadog recommends analyzing the performance metrics in the following ways:
```

> - **Dashboards**: they provide you with a bird-eye view of your application's performance. For example, the out-of-the-box [Performance Overview dashboard][1] can be filtered on [default attributes][2] collected by RUM to surface issues affecting only a subset of users. This dashboard can be cloned and customized to your specific needs. All [RUM performance metrics](#all-performance-metrics) can be used in dashboard queries.

Reviewer: ```suggestion
- **Dashboards**: provides you with a high-level view of your application's performance. For example, the out-of-the-box [Performance Overview dashboard][1] can be filtered on [default attributes][2] collected by RUM to surface issues impacting a subset of users. This dashboard can be cloned and customized to your specific needs. All [RUM performance metrics](#all-performance-metrics) can be used in dashboard queries.
```

- **RUM waterfall**: accessible for every single RUM view event in the [RUM Explorer][3], it lets you troubleshoot the performance of a specific page view. It shows how your website assets and resources, long tasks, and frontend errors affect performance for your end users, at the page level.

### Core Web Vitals

> [Google's Core Web Vitals][4] are a set of three metrics designed to monitor a site's user experience. These metrics, which focus on load performance, interactivity, and visual stability, give you an overview of what matters the most to your users. Each metric comes with guidance on the range of values that translate to good user experience. We recommend monitoring the 75th percentile for these metrics.

Reviewer: ```suggestion
[Google's Core Web Vitals][4] are a set of three metrics designed to monitor a site's user experience. These metrics focus on giving you a view of load performance, interactivity, and visual stability. Each metric comes with guidance on the range of values that translate to good user experience. Datadog recommends monitoring the 75th percentile for these metrics.
```

{{< img src="real_user_monitoring/browser/core-web-vitals.png" alt="Core Web Vitals summary visualization"  >}}

| Metric                   | Focus            | Description                                                                                           | Target value |
|--------------------------|------------------|-------------------------------------------------------------------------------------------------------|--------------|
| [Largest Contentful Paint][5] | Load performance | Moment in the page load timeline in which the largest DOM object in the viewport (i.e. visible on screen) is rendered.         | <2.5s       |
| [First Input Delay][6]        | Interactivity    | Time elapsed between a user’s first interaction with the page and the browser’s response.             | <100ms      |
| [Cumulative Layout Shift][7]  | Visual stability | Quantifies unexpected page movement due to dynamically loaded content (for example, third-party ads) where 0 means no shifts happening. | <0.1        |

**Note**: Metrics collected from your real users can differ from those calculated for pages loaded in a synthetic environment.

### All performance metrics

> RUM view performance metrics are collected from the [Navigation Timing API][8], the [Paint Timing API][9], the [web-vitals library][10] or in-house algorithms.

Author: * Would "in-house implementation of the web-vitals library" be better?
* There are already links to the web.dev docs for each metric. IMO this content would not be understood by 99% of our users (just like Navigation Timing API and similar links on the page to be honest...). What do you think?

> RUM view performance metrics are collected from the [Navigation Timing API][8], the [Paint Timing API][9], the [web-vitals library][10] or in-house algorithms.

Reviewer 2: > Would "in-house implementation of the web-vitals library" be better?

Do we care about how we have implemented web vitals?

> There are already links to the web.dev docs for each metric. IMO this content would not be understood by 99% of our users (just like Navigation Timing API and similar links on the page to be honest...). What do you think?

agree, IMO, Links to spec could be only added in metric description.

FMU the intention was to inform that some of the metric are "standard" and other are in house.
If that is right, maybe we could make the distinction clearer by having two metrics table, one for standard browser metrics and another for in house metrics. 
wdyt?

> RUM view performance metrics are collected from the [Navigation Timing API][8], the [Paint Timing API][9], the [web-vitals library][10] or in-house algorithms.

Author: You're right, that's useless, I'll just delete that sentence altogether. We'll probably remove this list of all metrics and just keep it in the data collected section, so no need for separated tables on this page 🔥

| `view.largest_contentful_paint` | number (ns) | Moment in the page load timeline in which the largest DOM object in the viewport (i.e. visible on screen) is rendered.                                                                                                |
| `view.first_input_delay`        | number (ns) | Time elapsed between a user’s first interaction with the page and the browser’s response.                                                                                                                             |
| `view.cumulative_layout_shift`  | number      | Quantifies unexpected page movement due to dynamically loaded content (for example, third-party ads) where 0 means no shifts happening.                                                                               |
| `view.loading_time`             | number (ns) | Time until the page is ready and no network request or DOM mutation is currently occurring. [More info](#how-is-loading-time-calculated).                                                                             |

> | `view.first_contentful_paint`   | number (ns) | Time when the browser first renders any text, image (including background images), non-white canvas, or SVG. For more information about browser rendering, see the [w3 definition][11].                               |

Reviewer 2: ```suggestion
| `view.first_contentful_paint`   | number (ns) | Time when the browser first renders any text, image (including background images), non-white canvas, or SVG. For more information about browser rendering, see the [w3c definition][11].                               |
```
